### ISSUE TITLE
Unable to navigate to historic version

### ISSUE THREAD

**Issue by takesson** [Reposoft/rweb#22]
Both via details page or the full history page in 1.6.4.

`Runtime error (code 8): Undefined variable: targetUrl `

**Comment by solsson** [comment]
Apparently this works in Docker containers. I will look into it.

**Comment by solsson** [comment]
@takesson Do you get a stack trace in view source? Or a line number in logs?

**Comment by takesson** [comment]
Introduced btw 1.6.2 and 1.6.3.
```
<!-- Error level 8. Stack trace:
file	line	function
/srv/cms/rweb/repos-web/open/SvnOpenFile.class.php	741	reportError
/srv/cms/rweb/repos-web/open/SvnOpenFile.class.php	510	getMimeTypePropertyValue
/srv/cms/rweb/repos-web/open/SvnOpenFile.class.php	555	getType
/tmp/rL3Nydi9jbXMvcndlYi9yZXBvcy13ZWI/smarty-cache/templates_c/%%49^49B^49B134AA%%index_en.html.php	105	isHtml
/srv/cms/rweb/repos-web/lib/smarty/libs/Smarty.class.php	1257	include
/srv/cms/rweb/repos-web/lib/smarty/libs/Smarty.class.php	1107	fetch
/srv/cms/rweb/repos-web/conf/Presentation.class.php	331	display
/srv/cms/rweb/repos-web/open/index.php	66	display
 -->```

**Comment by takesson** [comment]
History works for folders but not for files.

**Comment by solsson** [comment]
The syntax for that error message must have been broken for a long time. I've fixed it in 606d47fa but I still can't reproduce that issue. I suspect it's an error with peg rev handling.

For example I create a text file http://svn/repos-web/open/?base=test&target=%2Frepos.txt then go through history to "details for rev X" http://svn/repos-web/open/?base=test&target=/repos.txt&p=8&r=8&rweb=details then delete the file and reload the details page for the now historical revision, mime-type changes from `text/plain` to `application/x-unknown`. That's because the [propget](https://github.com/Reposoft/rweb/blob/v1.6.4/repos-web/open/SvnOpenFile.class.php#L521) returns something falsey so we continue there and return [unknown](https://github.com/Reposoft/rweb/blob/v1.6.4/repos-web/open/SvnOpenFile.class.php#L527), but if HEAD exists we get a proper value from Apache.

What I don't understand is why the `propget` fails instead of returning a non-value. I've verified that `@[peg]` is set properly in the command. The details page bails much earlier if the combination peg+rev is invalid or points to a nonexistent file.

@takesson Can you insert something like `print_r($cmd); exit;` before the [if ($cmd->getExitcode())](https://github.com/Reposoft/rweb/blob/v1.6.4/repos-web/open/SvnOpenFile.class.php#L752) on line 752 to see the command and the result? I've verified that if there is an svn:mime-type property on the historical revision, we do get that value.

**Comment by takesson** [comment]
Interesting to note, viewing historical on a rev that happens to still be HEAD works (all older revs fail even if there has been no delete/move).

```
SvnOpen Object ( [command] => Command Object ( [operation] => svn [args] => Array ( [0] => --username="takesson" --password="..." --no-auth-cache --non-interactive --config-dir "/tmp/rL3Nydi9jbXMvcndlYi9yZXBvcy13ZWI/svn-config-dir" [1] => propget [2] => svn:mime-type [3] => "http://pds-svn-gbg.pdsvision.net/svn/demo1/vvab/xml/documents/900276.xml@174" ) [output] => Array ( [0] => svn: warning: W200017: Property 'svn:mime-type' not found on 'http://pds-svn-gbg.pdsvision.net/svn/demo1/vvab/xml/documents/900276.xml@174' [1] => svn: E200000: A problem occurred; see other errors for details ) [exitcode] => 1 ) [operation] => propget ) 
```

**Comment by solsson** [comment]
This seems to be an interesting API change in in svn 1.9:
```
$ docker run --rm solsson/svnclient:latest svn propget svn:mime-type https://svn.apache.org/repos/asf/subversion/tags/1.9.5/README
$ echo $?
0
$ docker run --rm solsson/svnclient:1.9    svn propget svn:mime-type https://svn.apache.org/repos/asf/subversion/tags/1.9.5/README
svn: warning: W200017: Property 'svn:mime-type' not found on 'https://svn.apache.org/repos/asf/subversion/tags/1.9.5/README'
svn: E200000: A problem occurred; see other errors for details
$ echo $?
1
```

I've added support for svn 1.9 in 652d82ef. Haven't had the chance to test it properly except with the output from above. Relased v1.6.5 in 4c361a4c.

**Comment by solsson** [comment]
If you can confirm the fix I will apply it to 1.7.x as well.

I still suspect there is peg/rev issues with this function, but it should not matter much in practice because `svn:mime-type` is rarely edited.

**Comment by takesson** [comment]
Works on 1.9.3. Thanks.
